drivers/sdhc/rcar_mmc.c

```diff
@@ -744,72 +744,81 @@ static int rcar_mmc_request(const struct device *dev,
 	struct sdhc_command *cmd,
 	struct sdhc_data *data)
 {
-	int ret;
+	int ret = -ENOTSUP;
 	uint32_t reg;
 	uint32_t response_type;
 	bool is_read = true;
+	int retries;
 
 	if (!dev || !cmd) {
 		return -EINVAL;
 	}
 
-	/* TODO: add retries mechanism, use retries counter from cmd for that */
 	response_type = cmd->response_type & SDHC_NATIVE_RESPONSE_MASK;
+	retries = cmd->retries ? : 1;
```

Comment:
This does not look right. In my understanding, `retries` tells how many times you should retry operation if it fails. So, for example, `cmd->retries == 0` tells that you should not retry operation at all and `cmd->retries == 1` tells you that you should try one more time. At least, this is how all other SDHC drivers behave. 

But your code makes this two cases equal.

Reply:
done